packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -50,6 +52,22 @@ export abstract class AbstractResourcePreferenceProvider implements PreferencePr
         return this.preferences;
     }
 
+    async setPreference(key: string, value: string): Promise<void> {
+        const resource = await this.resource;
+        if (resource.saveContents) {
+            let preferences = this.preferences;
+            if (preferences) {
+                preferences[key] = value;
+            } else {
+                preferences = {[key]: value};
+            }
+            const content = await resource.readContents();
+            const indent = detectIndent(content).ident || '   ';
+            await resource.saveContents(JSON.stringify(preferences, undefined, indent));
+            this.onDidPreferencesChangedEmitter.fire(undefined);
```

Comment:
wouldn't the changes be picked up by `readPreferences`?

Reply:
The file resource won't fire a notification on save it assumes that the client already has up to date state.